app/lib/actions/katello/product/create.rb

```diff
@@ -2,33 +2,44 @@ module Actions
   module Katello
     module Product
       class Create < Actions::EntryAction
-        def plan(product, organization)
+        def plan(product, organization, subscription_start = nil)
           sequence do
             product.provider = organization.anonymous_provider
             product.organization = organization
+            product.setup_label_from_name
+            product.cp_id = unused_product_id(organization.label, product.label)
+            product.save!
 
-            cp_create = plan_action(::Actions::Candlepin::Product::Create,
-                                    :owner => product.organization.label,
-                                    :name => product.name,
-                                    :multiplier => 1,
-                                    :attributes => [{:name => "arch", :value => "ALL"}])
-
-            cp_id = cp_create.output[:response][:id]
+            plan_action(::Actions::Candlepin::Product::Create,
+                        :owner => product.organization.label,
+                        :name => product.name,
+                        :id => product.cp_id,
+                        :multiplier => 1,
+                        :attributes => [{:name => "arch", :value => "ALL"}])
 
             sub_create = plan_action(::Actions::Candlepin::Product::CreateUnlimitedSubscription,
                         :owner_key => organization.label,
-                        :product_id => cp_id)
+                        :product_id => product.cp_id,
+                        :start_time => subscription_start)
 
             subscription_id = sub_create.output[:response][:id]
 
-            product.save!
-            action_subject product, :cp_id => cp_id
+            action_subject product, :cp_id => product.cp_id
 
             plan_self
             plan_action Katello::Product::ReindexSubscriptions, product, subscription_id
           end
         end
 
+        def unused_product_id(owner, product_label)
+          id = ::Katello::Util::Data.md5hash("#{owner}-#{product_label}")
+          if ::Katello::Product.find_by(:cp_id => id)
+            SecureRandom.random_number(999_999_999_999)
```

Comment:
@jlsherrill I'll leave it up to your judgement - It would just strike me as odd to see the two different types of cp_ids. Another thought is to add product + label + number that increments until a unique one is found
i.e.
foo_bar --> exists
foo_bar_1 --> exists
foo_bar_2 --> does not exist, use this as cp_id

where foo is owner and bar is product_label